packages/angular_devkit/benchmark/src/main.ts

```diff
@@ -0,0 +1,182 @@
+}: MainOptions): Promise<0 | 1> {
+
+  // Show usage of the CLI tool, and exit the process.
+  function usage(logger: logging.Logger) {
+    logger.info(tags.stripIndent`
+    benchmark [options] -- [command to benchmark]
+
+    Collects process stats from running the command.
+
+    Options:
+        --help                    Show this message.
+        --verbose                 Show more information while running.
+        --exit-code               Expected exit code for the command. Default is 0.
+        --iterations              Number of iterations to run the benchmark over. Default is 5.
+        --retries                 Number of times to retry when process fails. Default is 5.
+        --cwd                     Current working directory to run the process in.
+        --output-file             File to output benchmark log to.
+        --overwrite-output-file   If the output file should be overwritten rather than appended to.
+        --prefix                  Logging prefix.
+
+    Example:
+        benchmark --iterations=3 -- node my-script.js
+  `);
+  }
+
+  interface BenchmarkCliArgv {
+    help: boolean;
+    verbose: boolean;
+    'overwrite-output-file': boolean;
+    'exit-code': number;
+    iterations: number;
+    retries: number;
+    'output-file': string | null;
+    cwd: string;
+    prefix: string;
+    '--': string[] | null;
+  }
+
+  // Parse the command line.
+  const argv = minimist(args, {
+    boolean: ['help', 'verbose', 'overwrite-output-file'],
+    default: {
+      'exit-code': 0,
+      'iterations': 5,
+      'retries': 5,
+      'output-file': null,
+      'cwd': process.cwd(),
+      'prefix': '[benchmark]',
+    },
+    '--': true,
+  }) as {} as BenchmarkCliArgv;
+
+  // Create the DevKit Logger used through the CLI.
+  const logger = new logging.TransformLogger(
+    'benchmark-prefix-logger',
+    stream => stream.pipe(map(entry => {
+      if (argv['prefix']) { entry.message = `${argv['prefix']} ${entry.message}`; }
+
+      return entry;
+    })),
+  );
+
+  // Log to console.
+  logger
+    .pipe(filter(entry => (entry.level != 'debug' || argv['verbose'])))
+    .subscribe(entry => {
+      let color: (s: string) => string = x => terminal.dim(terminal.white(x));
+      let output = stdout;
+      switch (entry.level) {
+        case 'info':
+          color = terminal.white;
+          break;
+        case 'warn':
+          color = terminal.yellow;
+          break;
+        case 'error':
+          color = terminal.red;
+          output = stderr;
+          break;
+        case 'fatal':
+          color = (x: string) => terminal.bold(terminal.red(x));
+          output = stderr;
+          break;
+      }
+
+      output.write(color(entry.message) + '\n');
+    });
+
+
+  // Print help.
+  if (argv['help']) {
+    usage(logger);
+
+    return 0;
+  }
+
+  // Exit early if we can't find the command to benchmark.
+  if (!Array.isArray(argv['--']) || (argv['--'] as Array<string>).length < 1) {
```

Comment:
nit: I'm surprised you need to cast `['--']` as a `string[]`

Reply:
It is not `['--']` that is being cast but rather `argv['--']`. Depending on the content of `process.argv`, that array might not exist.

Looking back on this logic I think I'm casting it way more than needed. Will make an intermediate var that's properly typed.